cumulus/zombienet/zombienet-sdk/tests/elastic_scaling/elastic_scaling_multiple_blocks_per_slot.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+
+use crate::helpers::{
+	assert_finalized_block_height, assert_para_throughput, create_assign_core_call,
+};
+use polkadot_primitives::Id as ParaId;
+use serde_json::json;
+use subxt::{OnlineClient, PolkadotConfig};
+use subxt_signer::sr25519::dev;
+use zombienet_sdk::{NetworkConfig, NetworkConfigBuilder};
+
+const PARA_ID: u32 = 2400;
+
+/// This test spawns a parachain network.
+/// Initially, one core is assigned. We expect the parachain to produce 1 block per relay.
+/// As we increase the number of cores via `assign_core`, we expect the block pace to increase too.
+/// **Note:** The runtime in use here has 6s slot duration, so multiple blocks will be produced per
+/// slot.
+#[tokio::test(flavor = "multi_thread")]
+async fn elastic_scaling_multiple_block_per_slot() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	let config = build_network_config().await?;
+
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+	let network = spawn_fn(config).await?;
+
+	let relay_node = network.get_node("validator-0")?;
+	let para_node_elastic = network.get_node("collator-1")?;
+
+	let relay_client: OnlineClient<PolkadotConfig> = relay_node.wait_client().await?;
+	let alice = dev::alice();
+	assert_para_throughput(
+		&relay_client,
+		10,
+		[(ParaId::from(PARA_ID), 8..11)].into_iter().collect(),
+	)
+	.await?;
+
+	let assign_cores_call = create_assign_core_call(&[2, 3], PARA_ID);
+
+	relay_client
+		.tx()
+		.sign_and_submit_then_watch_default(&assign_cores_call, &alice)
+		.await
+		.inspect(|_| log::info!("Tx send, waiting for finalization"))?
+		.wait_for_finalized_success()
+		.await?;
+	log::info!("2 more cores assigned to each parachain");
+
+	assert_para_throughput(
+		&relay_client,
+		15,
+		[(ParaId::from(PARA_ID), 39..46)].into_iter().collect(),
```

Comment:
dq: in ideal world `15*3` (15 blocks, 3 cores) would be expected? Do I read it correctly?